resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -59,5 +66,28 @@ private[spark] object KubernetesUtils {
     }
   }
 
+  def loadPodFromTemplate(
+      kubernetesClient: KubernetesClient,
+      templateFile: File): SparkPod = {
+    try {
+      val pod = kubernetesClient.pods().load(templateFile).get()
+      pod.getSpec.getContainers.asScala.toList match {
```

Comment:
I think you can use `headOption` here to get an `Option` object and use `map`, `getOrElse`.

Reply:
Ah actually because we're extracting first and rest this implementation is probably cleaner.